bazaar/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2019 Alain Brenzikofer
+// This file is part of Encointer
+//
+// Encointer is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Encointer is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Encointer.  If not, see <http://www.gnu.org/licenses/>.
+
+
+//! # Encointer Bazaar Module
+//!
+//! provides functionality for
+//! - creating new bazaar entries (shop and articles)
+//! - removing existing entries (shop and articles)
+//!
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+#[macro_use]
+extern crate approx;
+
+use frame_support::{
+    decl_event, decl_module, decl_storage, decl_error,
+    dispatch::DispatchResult,
+    ensure,
+    storage::{StorageDoubleMap, StorageMap},
+    weights::{DispatchClass, Pays},
+    debug
+};
+use rstd::prelude::*;
+use frame_system::ensure_signed;
+use codec::{Decode, Encode};
+
+use encointer_currencies::{CurrencyIdentifier};
+
+pub trait Trait: frame_system::Trait 
+    + encointer_currencies::Trait 
+{
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+}
+
+pub type ShopIdentifier = u64; 
```

Comment:
I doubt this is the right branch. Shouldn't ShopIdentifier be an IPFS hash?

Reply:
Sorry, you're right. I will update the dependencies and make a new PR.